package/MDAnalysis/coordinates/PDB.py

```diff
@@ -1107,13 +1114,15 @@ def get_attr(attrname, default):
             vals['name'] = self._deduce_PDB_atom_name(atomnames[i], resnames[i])
             vals['altLoc'] = altlocs[i][:1]
             vals['resName'] = resnames[i][:4]
-            vals['chainID'] = segids[i][-1:]
             vals['resSeq'] = util.ltruncate_int(resids[i], 4)
             vals['iCode'] = icodes[i][:1]
             vals['pos'] = pos[i]  # don't take off atom so conversion works
             vals['occupancy'] = occupancies[i]
             vals['tempFactor'] = tempfactors[i]
             vals['segID'] = segids[i][:4]
+            vals['chainID'] = (
+                chainids[i][:4] if (chainids[i][:4] != ' ') else segids[i][-1:]
```

Comment:
My concerns regarding the 4-letter chainIDs and the confusion between empty and missing chainIDs is relevant here, too.